dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataAccessManager.kt

```diff
@@ -0,0 +1,179 @@
+                "The user $userId cannot access the dataset for the company $companyId, for the data type " +
+                    "$dataType and the reporting period $reportingPeriod. The dataset may not exists.",
+            )
+        }
+    }
+
+    /**
+     * This method is used to request access to a private dataset
+     * @param userId the userId of the user requesting access to the dataset
+     * @param companyId the companyId of the company to which the dataset belongs
+     * @param dataType the datatype of the dataset to which access was requested
+     * @param reportingPeriod the reportingPeriod of the dataset to which access was requested
+     */
+    fun createAccessRequestToPrivateDataset(
+        userId: String,
+        companyId: String,
+        dataType: DataTypeEnum,
+        reportingPeriod: String,
+        contacts: Set<String>?,
+        message: String?,
+    ) {
+        val existingRequestsOfUser = dataRequestRepository
+            .findByUserIdAndDatalandCompanyIdAndDataTypeAndReportingPeriod(
+                userId, companyId, dataType.name, reportingPeriod,
+            )
+        if (!existingRequestsOfUser.isNullOrEmpty()) {
+            val dataRequestEntity = existingRequestsOfUser[0]
+
+            val modificationTime = Instant.now().toEpochMilli()
+            dataRequestEntity.lastModifiedDate = modificationTime
+            dataRequestRepository.save(dataRequestEntity)
+            // TODO discuss if declined should be removed from this condition
+            if (dataRequestEntity.accessStatus == AccessStatus.Revoked || dataRequestEntity.accessStatus ==
+                AccessStatus.Declined || dataRequestEntity.accessStatus == AccessStatus.Public
+            ) {
+                dataRequestProcessingUtils.addNewRequestStatusToHistory(
+                    dataRequestEntity, dataRequestEntity.requestStatus, AccessStatus.Pending, modificationTime,
+                )
+
+                accessRequestLogger.logMessageForPatchingAccessStatus(
+                    dataRequestEntity.dataRequestId, AccessStatus.Pending,
+                )
+            }
+        } else {
+            storeAccessRequestEntityAsPending(
+                companyId, dataType, reportingPeriod,
+                contacts.takeIf { !it.isNullOrEmpty() },
+                message.takeIf { !it.isNullOrBlank() },
+            )
+        }
+    }
+
+    /**
+     * Stores a DataRequestEntity from all necessary parameters
+     * @param datalandCompanyId the companyID in Dataland
+     * @param dataType the enum entry corresponding to the framework
+     * @param reportingPeriod the reporting period
+     * @param contacts a list of email addresses to inform about the potentially stored data request
+     * @param message a message to equip the notification with
+     */
+    fun storeAccessRequestEntityAsPending(
+        datalandCompanyId: String,
```

Comment:
most times it's "companyId", isn't it?

Reply:
Add to ticket for findings for refactoring of community manager service